docs/09_tutorials/10_nested_containers.md

```diff
@@ -0,0 +1,199 @@
+
+### About *std::pair* And *std::map* In Cleos Commands
+
+In EOSIO code base, `std::pair` is a struct of fields *first* and *second*, `std::map` is regarded as an array/vector of structs with fields *key* and *value*, as observed in *.abi* files where `std::pair<uint16_t, uint16_t>` has *.abi* type
+`pair_uint16_uint16`, and `std::map<uint16_t, uint16_t>` has *.abi* type `pair_uint16_uint16[]`.
+
+However a container which is a combination of both `std::pair` and `std::map` may have a complex mixture of *key* and *value*, *first* and *second* which are documented in the *.sh* scripts, and in the comments of the *.cpp* files. Please pay special attention on the cleos commands of *setmm* for map of maps, *setmp* for map of pairs, *setpm* for pair of maps, and *setpp* for pair of pairs, because **the cleos commands of all these 4 actions use *"key"* and *"value"* with the *"value"* part embracing *"first"* and *"second"***. 
+
+For example, *setm* only uses *key* and *value*, and *setp* only uses *first* and *second*, but *setmm* and *setpp* use *"key"* and *"value"* with the *"value"* part embracing *"first"* and *"second"* as below:
+
+```sh
+cleos push action nestcontn2kv setmm '[1, [{"key":10,"value":[{"first":200,"second":2000}, {"first":201,"second":2001}] }, {"key":11,"value":[{"first":300,"second":3000}, {"first":301,"second":3001}] } ]]' -p alice@active
+
+cleos push action nestcontn2kv setpp '[1, {"key":30, "value":{"first":301, "second":302} }]' -p alice@active
+```
+
+The cleos commands of *setmp* and *setpm* follow the same pattern as shown in the *.sh* scripts of this project. Also, you should use it in the cleos commands whenever `std::pair` or `std::map` is nested with any other container such as *setvm* for vector of maps, *setvp* for vector of pairs etc., which are all documented in the two *.sh* scripts.
+
+## Supported Nested-Containers
+
+The following table describes which nested-containers combinations of STL set, vector, optional, map, pair and tuple in the std namespace are supported by current EOSIO implementation:
+
+|     | set | vector | optional | map | pair | tuple |
+| --- | --- | ---    | ---      | --- | ---  | ---   |
+| set | supported | supported   | **Not supported** | supported | supported  | supported  |
+| vector | supported | supported   | **Not supported** | supported | supported  | supported  |
+| optional | supported | supported   | supported | supported | supported  | supported  |
+| map | supported | supported   | supported | supported | supported  | supported  |
+| pair | supported | supported   | supported | supported | supported  | supported  |
+| tuple |  supported |  supported   |  supported |  supported |  supported  | supported  |
+
```

Comment:
set<optional> and vector<optional> will be supported since Farhad's code is already in the eos/ [develop] branch